read-fonts/src/tables/postscript/charstring.rs

```diff
@@ -78,6 +78,179 @@ where
     }
 }
 
+/// Command sink adapter that applies a scaling factor.
+///
+/// This assumes a 26.6 scaling factor packed into a Fixed and thus,
+/// this is not public and exists only to match FreeType's exact
+/// scaling process.
+pub(crate) struct ScalingSink26Dot6<'a, S> {
+    inner: &'a mut S,
+    scale: Fixed,
+}
+
+impl<'a, S> ScalingSink26Dot6<'a, S> {
+    pub fn new(sink: &'a mut S, scale: Fixed) -> Self {
+        Self { scale, inner: sink }
+    }
+
+    fn scale(&self, coord: Fixed) -> Fixed {
+        if self.scale != Fixed::ONE {
+            // The following dance is necessary to exactly match FreeType's
+            // application of scaling factors:
+            // 1. Multiply by 1/64
+            // <https://gitlab.freedesktop.org/freetype/freetype/-/blob/80a507a6b8e3d2906ad2c8ba69329bd2fb2a85ef/src/psaux/psft.c#L284>
+            let a = coord * Fixed::from_bits(0x0400);
+            // 2. Convert to 26.6 by truncation
+            // <https://gitlab.freedesktop.org/freetype/freetype/-/blob/80a507a6b8e3d2906ad2c8ba69329bd2fb2a85ef/src/psaux/psobjs.c#L2219>
+            let b = Fixed::from_bits(a.to_bits() >> 10);
+            // 3. Multiply by the original scale factor
+            // <https://gitlab.freedesktop.org/freetype/freetype/-/blob/80a507a6b8e3d2906ad2c8ba69329bd2fb2a85ef/src/cff/cffgload.c#L721>
+            let c = b * self.scale;
+            // Finally, we convert back to 16.16
+            Fixed::from_bits(c.to_bits() << 10)
+        } else {
+            // Otherwise, simply zero the low 10 bits
+            Fixed::from_bits(coord.to_bits() & !0x3FF)
+        }
+    }
+}
+
+impl<'a, S: CommandSink> CommandSink for ScalingSink26Dot6<'a, S> {
+    fn hstem(&mut self, y: Fixed, dy: Fixed) {
+        self.inner.hstem(y, dy);
+    }
+
+    fn vstem(&mut self, x: Fixed, dx: Fixed) {
+        self.inner.vstem(x, dx);
+    }
+
+    fn hint_mask(&mut self, mask: &[u8]) {
+        self.inner.hint_mask(mask);
+    }
+
+    fn counter_mask(&mut self, mask: &[u8]) {
+        self.inner.counter_mask(mask);
+    }
+
+    fn move_to(&mut self, x: Fixed, y: Fixed) {
+        self.inner.move_to(self.scale(x), self.scale(y));
+    }
+
+    fn line_to(&mut self, x: Fixed, y: Fixed) {
+        self.inner.line_to(self.scale(x), self.scale(y));
+    }
+
+    fn curve_to(&mut self, cx1: Fixed, cy1: Fixed, cx2: Fixed, cy2: Fixed, x: Fixed, y: Fixed) {
+        self.inner.curve_to(
+            self.scale(cx1),
+            self.scale(cy1),
+            self.scale(cx2),
+            self.scale(cy2),
+            self.scale(x),
+            self.scale(y),
+        );
+    }
+
+    fn close(&mut self) {
+        self.inner.close();
+    }
+}
+
+/// Command sink adapter that simplifies path operations.
+///
+/// This currently removes degenerate moves and lines.
```

Comment:
Comment would be more helpful if it offered:

* Why? 
* Do we think it might change in future? If so ... why?

Reply:
Why only moves and lines, why don't we filter all commands whose points are coincident? That pattern could also inform a more informative name?

Reply:
Like so many other things, it’s just what FT does so we follow.